setup.py

```diff
@@ -1,335 +1,397 @@
-            shutil.rmtree(dest_dir)
-        os.makedirs(dest_dir, exist_ok=True)
-
-        for test_file in os.listdir(src_test_dir):
-            src_file = os.path.join(src_test_dir, test_file)
-            dest_file = os.path.join(dest_dir, test_file)
-            if os.path.isfile(src_file):
-                if os.path.islink(dest_file) or os.path.isfile(dest_file):
-                    os.unlink(dest_file)
-                elif os.path.isdir(dest_file):
-                    shutil.rmtree(dest_file)
-
-                os.makedirs(os.path.dirname(dest_file), exist_ok=True)
-                os.symlink(src_file, dest_file)
-
-    def process_directory(self, root: str, dirs: list, files: list) -> None:
-        """Process a directory for modules and tests."""
-        module_file = None
-        if "gnomod.toml" in files:
-            module_file = os.path.join(root, "gnomod.toml")
-        elif "gno.mod" in files:
-            raise ValueError(
-                "gno.mod format is outdated. Please use gnomod.toml instead."
-            )
-
-        if module_file:
-            module_path = self.module_manager.extract_module_path(module_file)
-            if module_info := self.module_manager.get_module_info(root, module_path):
-                self.copy_module(module_info)
-
-
-def clone_repository(workdir: str) -> None:
-    """Clone the GnoSwap repository."""
-    os.chdir(workdir)
-    subprocess.run(
-        ["git", "clone", "https://github.com/gnoswap-labs/gnoswap.git"], check=True
-    )
-    os.chdir("gnoswap")
+
+class IntegrationTest(NamedTuple):
+    """Integration test file information."""
+
+    source_path: Path
+    converted_name: str
 
 
-def setup_contracts(workdir: str, exclude_tests: bool = False) -> None:
-    """Set up all contracts and tests."""
-    module_manager = GnoModuleManager(workdir)
-    copier = ContractCopier(module_manager, exclude_tests=exclude_tests)
+@dataclass(frozen=True)
+class PathConfig:
+    """Configuration for directory paths."""
 
-    for root, dirs, files in os.walk("contract"):
-        copier.process_directory(root, dirs, files)
+    workdir: Path
+    gno_examples_dir: Path
+    gno_root_dir: Path
+
+    @classmethod
+    def from_workdir(cls, workdir: Path) -> PathConfig:
+        """Create `PathConfig` from a working directory."""
+        return cls(
+            workdir=workdir,
+            # If the gno repository name is different, change only `gno` in this path to the appropriate one.
+            # For example, if the cloned repository is `gno-core/...`, replace it with `gno-core`.
+            gno_examples_dir=workdir / "gno" / "examples" / "gno.land",
+            gno_root_dir=workdir / "gno" / "gno.land",
+        )
 
-    for root, dirs, files in os.walk("tests/scenario"):
-        copier.process_directory(root, dirs, files)
 
+def parse_module_from_toml(data: dict) -> Optional[str]:
+    """Parse module path from toml data"""
+    return data.get("module")
```

Comment:
This function is too simple and only called once. Consider inlining it directly in `read_module_path()`: `return data.get("module")`